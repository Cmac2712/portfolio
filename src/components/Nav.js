import React from 'react'
import posed from 'react-pose'

const NavPosed = posed.nav({
	active: {
		applyAtStart: { opacity: 0 }, 
		opacity: 1
	},
	inactive: {
		applyAtStart: { opacity: 0 }, 
		opacity: 1
	}
})

class Nav extends React.Component {

	constructor(props) {
		super(props)

		this.handleClick = this.handleClick.bind(this)
	}

	componentDidMount() {
		// Avoid testing scroll-to as enzyme won't be able to render this component
		if (process.env.NODE_ENV === 'test') return;

		// We need 'require' here because we're using server-side rendering
		this.scrollTo = require('react-scroll-to-component')
	}

	handleClick(link) {
		this.props.changeSection(link.href)
	}

	render () {
		return (
			<div className='nav-container'>
				<NavPosed pose={this.props.isVisible ? 'inactive' : 'active'}>
					<nav className={this.props.isVisible ? "main-nav" : "main-nav main-nav--fixed"}>
							<div className='wrap'>
								{this.props.links.map((link, i) => {
									return (
										<a className={this.props.currentSection === link.name.toLowerCase() ? "main-nav__link active" : "main-nav__link"}
										   key={i}
										   href={link.href}
											onClick={(event) => {
												event.preventDefault()
												this.handleClick(link)
											}}>
										   {link.name}
										</a>
									)
								})}
							</div>
					</nav>
				</NavPosed>
			</div>
		)
	}

}

export default Nav
