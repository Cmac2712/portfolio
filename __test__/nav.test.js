import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Nav from '../src/components/Nav';

const mockLinks = [
	{
		"name": "About", 
		"href": "#about"
	},
	{
		"name": "Skills", 
		"href": "#skills"
	},
	{
		"name": "Experience", 
		"href": "#experience"
	} 
];
const nav = <Nav links={mockLinks} currentSection="about" />
const wrapper = mount(nav);

describe('<Nav />', () => {

	it('should match snapshot', () => {
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it('should render 3 links', () => {
		expect(wrapper.find('a').length).toBe(3);
	});

	it('"about link should have "active" class when current section is set to "about""', () => {
		const firstLink = wrapper.find('a').first();

		expect(firstLink.hasClass('active')).toBe(true);
	})

});
