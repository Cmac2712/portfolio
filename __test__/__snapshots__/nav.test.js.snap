// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match snapshot 1`] = `
<Nav
  currentSection="about"
  links={
    Array [
      Object {
        "href": "#about",
        "name": "About",
      },
      Object {
        "href": "#skills",
        "name": "Skills",
      },
      Object {
        "href": "#experience",
        "name": "Experience",
      },
    ]
  }
>
  <div
    className="nav-container"
  >
    <ForwardRef
      pose="active"
    >
      <PoseElement
        elementType="nav"
        innerRef={null}
        pose="active"
        poseConfig={
          Object {
            "active": Object {
              "applyAtStart": Object {
                "opacity": 0,
              },
              "opacity": 1,
            },
            "inactive": Object {
              "applyAtStart": Object {
                "opacity": 0,
              },
              "opacity": 1,
            },
          }
        }
      >
        <nav>
          <nav
            className="main-nav main-nav--fixed"
          >
            <div
              className="wrap"
            >
              <a
                className="main-nav__link active"
                href="#about"
                key="0"
                onClick={[Function]}
              >
                About
              </a>
              <a
                className="main-nav__link"
                href="#skills"
                key="1"
                onClick={[Function]}
              >
                Skills
              </a>
              <a
                className="main-nav__link"
                href="#experience"
                key="2"
                onClick={[Function]}
              >
                Experience
              </a>
            </div>
          </nav>
        </nav>
      </PoseElement>
    </ForwardRef>
  </div>
</Nav>
`;
